clippy_lints/src/collect.rs

```diff
@@ -69,11 +75,11 @@ struct Suggestion {
 
 fn format_suggestion_pattern<'a, 'tcx>(
     cx: &LateContext<'a, 'tcx>,
-    collection_ty: &TypeVariants,
+    collection_ty: &TyKind<'_>,
```

Comment:
`TyKind` should never be used directly, you can write `Ty` instead.
@Manishearth @oli-obk I think clippy might have a general problem wrt this.